display: add support for AL_88 color format

Finomnis opened the pull request:
Matches LVGL's `LV_COLOR_FORMAT_AL88`.

Closes #94389.


-----

`west twister -p native_sim/native/64 -s drivers.display.read_write.sdl.al88 --fixture display` succeeds.

Several `mono10` and `mono01` tests still fail, but this is already reported in #88714 and unrelated to this change.

------

```
west build -b native_sim/native/64 samples/modules/lvgl/screen_transparency/ -t run -p auto
````
<img width="329" height="270" alt="image" src="https://github.com/user-attachments/assets/3b243f1a-9cdf-4007-8759-8729ad4d343e" />

-----
```
west build -b native_sim/native/64 samples/modules/lvgl/screen_transparency/ -t run -p auto -- \
    -DCONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_AL_88=y
````
<img width="327" height="268" alt="image" src="https://github.com/user-attachments/assets/9aa18c92-5583-4432-ad03-8e4d640cc16d" />

-----
```
west build -b native_sim/native/64 samples/drivers/display/ -t run -p auto -- \
    -DCONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_AL_88=y
````
<img width="326" height="268" alt="image" src="https://github.com/user-attachments/assets/3983f0ba-b8b8-4009-bac1-e1e607049536" />

Finomnis committed:
display: add PIXEL_FORMAT_AL_88

When rendering an overlay on top of something like a video stream on a
monochrome display, the only way to achieve this right now is by using
ARGB8888 and discarding two of the colors, which is not ideal.

This introduces the new AL_88 pixel format, which is much more efficient
for this usecase.

This change is somewhat inspired by LVGL, which also supports AL_88
natively.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

faxe1008 commented on samples/drivers/display/src/main.c:
```diff
@@ -165,6 +165,31 @@ static inline void fill_buffer_l_8(enum corner corner, uint8_t grey, uint8_t *bu
+static void fill_buffer_al_88(enum corner corner, uint8_t grey, uint8_t *buf,
+				 size_t buf_size)
+{
+	uint16_t color = 0;
```
Initialization not needed

Finomnis replied:
Doesn't hurt either, as the switch has no default case

kartben replied:
> Doesn't hurt either, as the switch has no default case

well, it should :)
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#:~:text=Every%20switch%20statement%20shall%20have%20a%20default%20label

faxe1008 commented on tests/drivers/display/display_check/src/main.c:
```diff
@@ -160,6 +160,31 @@ static inline void fill_buffer_l_8(enum corner corner, uint8_t grey, uint8_t *bu
+static void fill_buffer_al_88(enum corner corner, uint8_t grey, uint8_t *buf,
+				 size_t buf_size)
+{
+	uint16_t color = 0;
```
Same here

kartben replied:
ditto

Finomnis committed:
display: sdl: Add support for AL_88 color format

The SDL display driver can now handle and display AL_88 content.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

Finomnis committed:
samples: drivers: display: Add support for AL_88 color

Renders three different shades of gray in each corner, with one corner
continually changing color.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

Finomnis committed:
tests: drivers: display: Add test cases for AL_88

Enable display_read_write test for AL_88 color and add support for
AL_88 in display_check.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

Finomnis committed:
lvgl: Add support for AL_88 color

While both LVGL and Zephyr support AL_88 color, the glue to connect them
was still missing. Now AL_88 should be fully supported by Zephyr+LVGL.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

Finomnis committed:
samples: lvgl: Refactor screen_transparency sample

The LVGL screen_transparency sample was refactored in two ways:

- Remove old obsolete display initialization code. LVGL does not even
  react to this code because LVGL color format is determined during LVGL
  initialization, which comes before main().
  Further, the way it is written right now is incompatible with `AL_88`
  color.
- Add some RGB color text for demonstration purposes.

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

Finomnis committed:
display: sdl: fix clear color for ARGB8888

Clear color for ARGB8888 was set to opaque white, which is an odd
choice.

This change makes the clear color "transparent".

Signed-off-by: Martin Stumpf <finomnis@gmail.com>
